packages/containers-shared/tests/docker-context.test.ts

```diff
@@ -0,0 +1,65 @@
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import { resolveDockerHost } from "../src/utils";
+
+const mockedDockerContextLsOutput = `{"Current":true,"Description":"Current DOCKER_HOST based configuration","DockerEndpoint":"unix:///current/run/docker.sock","Error":"","Name":"default"}
+{"Current":false,"Description":"Docker Desktop","DockerEndpoint":"unix:///other/run/docker.sock","Error":"","Name":"desktop-linux"}`;
+
+vi.mock("node:child_process");
+
+describe.skipIf(process.platform !== "linux" && process.env.CI === "true")(
+	"resolveDockerHost",
+	() => {
+		let mockExecFileSync: ReturnType<typeof vi.fn>;
+
+		beforeEach(async () => {
+			vi.clearAllMocks();
+			const childProcess = await import("node:child_process");
+			mockExecFileSync = vi.mocked(childProcess.execFileSync);
+
+			mockExecFileSync.mockReturnValue(mockedDockerContextLsOutput);
+		});
+
+		afterEach(() => {
+			vi.unstubAllEnvs();
+		});
+
+		it("should return WRANGLER_DOCKER_HOST when set", async () => {
+			vi.stubEnv("WRANGLER_DOCKER_HOST", "unix:///foo/wrangler/socket");
+			vi.stubEnv("DOCKER_HOST", "unix:///bar/docker/socket");
+
+			const result = await resolveDockerHost("/no/op/docker");
+			expect(result).toBe("unix:///foo/wrangler/socket");
```

Comment:
2 nits:
- When I see "foo" l31, I have to go back l27/28 to see what it is vs seeing unix:///WRANGLER_DOCKER_HOST
- using `.resolves`
 
```suggestion
			vi.stubEnv("WRANGLER_DOCKER_HOST", "unix:///WRANGLER_DOCKER_HOST");
			vi.stubEnv("DOCKER_HOST", "unix:///DOCKER_HOST");

			await expect(resolveDockerHost("/no/op/docker")).resolves.toBe("unix:///WRANGLER_DOCKER_HOST");
```